File: 프로그래머스/1/135808. 과일 장수/과일 장수.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

int solution(int k, int m, vector<int> score) {
    int answer = 0;
    sort(score.rbegin(),score.rend());
    
    int sum = 0;
    int stack = 0;
    for(auto s : score)
    {
        if(s > k) continue;
        
        sum+=s;
        stack++;
        if(stack == m)
        {
            answer += s*m;
            stack = 0;
        }
    }
    
    return answer;
}

File: 프로그래머스/1/135808. 과일 장수/과일 장수.cs
using System;
using System.Linq;
public class Solution {
    public int solution(int k, int m, int[] score) {
        int answer = 0;
        Array.Sort(score);
        Array.Reverse(score);
        for (int i = 0; i < score.Length; i += m)
        {
            if(i+m > score.Length) break;
            int low = score[i];
            for (int j = i; j < i+m; j++)
            {
                if (score[j] < low) low = score[j];
            }
            answer += low * m;
        }
        return answer;
    }
}